### ISSUE TITLE
StateVersionOutput expects all outputs to be strings, and fails when other output types exist

### ISSUE THREAD

**Issue by sfc-gh-ahess** [hashicorp/go-tfe#158]
Terraform supports output types other than 'string'.  For example, this Terraform declaration produces an output type of 'array':

```
output "instance_ips" {
  value = aws_instance.ubuntu.*.private_ip
}
```

resulting output:
```
instance_ips: [
  "10.1.2.3",
  "10.1.2.4",
  "10.1.2.5" 
]
```

This is why the State Output contains a "type" field, which will be "array" in this case.

However, the tfe-go library defines the State Version Output as:
```
type StateVersionOutput struct {
	ID        string `jsonapi:"primary,state-version-outputs"`
	Name      string `jsonapi:"attr,name"`
	Sensitive bool   `jsonapi:"attr,sensitive"`
	Type      string `jsonapi:"attr,type"`
	Value     string `jsonapi:"attr,value"`
}
```

Note that Value is 'string' only. 

When .Read is called on an Output of type 'array', the Read call will fail with an error, as the jsonapi is not able to unmarshal the array value into this struct.

Here's an example API response from Terraform with an array:

```
{
  "data": {
    "id": "wsout-abcdef",
    "type": "state-version-outputs",
    "attributes": {
      "name": "instance_ips",
      "sensitive": false,
      "type": "array",
      "value": [
        "10.180.69.26",
        "10.180.69.11",
        "10.180.69.37"
      ]
    },
    "links": {
      "self": "/api/v2/state-version-outputs/wsout-abcdef"
    }
  }
}
```

Thanks!
-alex

**Comment by rorsten** [comment]
I'm hitting this as well, with a state output that's an integer. And I can think of some workspaces where the outputs are maps. IIRC in the case of maps there's also a schema field? I would be quite satisfied with getting something from cty or even a json.RawMessage of the field.

**Comment by sfc-gh-ahess** [comment]
I've fixed this in a fork of the repo, you can see my PR here:
https://github.com/hashicorp/go-tfe/pull/161

**Comment by omarismail** [comment]
Hey @sfc-gh-ahess, we know this has been a long standing issue and apologies for the delay. We have addressed this problem in #246. Now the type of `StateVersionOutput.Value` is an `interface{}` and you can type cast it into anything you'd like.
